tc-server/src/main/java/com/tc/objectserver/entity/ServerEntityFactory.java

```diff
@@ -92,7 +93,15 @@ public ServerEntityFactory(ServiceLocator loader) {
       }
     }
     return msgs;
-  }  
+  }
+
+  public boolean isPermanentEntity(EntityID entity) {
+    Predicate<EntityID> sameName = e -> e.getEntityName().equals(entity.getEntityName());
+    Predicate<EntityID> sameClass = e -> e.getClassName().equals(entity.getClassName());
+    return getAnnotatedEntities().stream()
+        .map(p -> p.getEntityDescriptor().getEntityID())
+        .anyMatch(sameName.and(sameClass));
+  }
 
```

Comment:
should getAnnotatedEntities() be built once and held in a map/list since it is now being queried on every entity create?

Reply:
Is the complete set of AnnotatedEntities populated only once (e.g. at server startup)?  Or do additional ones get added during a server's lifetime?

Reply:
It's a classpath scan so it should not change in the life of the server.

Reply:
done